src/test/java/tests/AbstractTest.java

```diff
@@ -0,0 +1,97 @@
+package tests;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestExecutionExceptionHandler;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.ie.InternetExplorerDriver;
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+public abstract class AbstractTest implements TestExecutionExceptionHandler {
+
+    protected WebDriver driver;
+    protected Properties envProperties;
+
+    @BeforeEach
+    public void setUp() {
+        configureDriver();
+        envProperties = loadProperties(".env-test");
+    }
+
+    private Properties loadProperties(String filePath) {
+        Properties properties = new Properties();
+        try (FileInputStream fileInputStream = new FileInputStream(filePath)) {
+            properties.load(fileInputStream);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return properties;
+    }
+
+    private void configureDriver() {
+        Properties testProperties = loadProperties("src/test/resources/test.properties");
+
+        String browser = testProperties.getProperty("browser", "chrome").toLowerCase();
+        switch (browser) {
+            case "firefox":
+                WebDriverManager.firefoxdriver().setup();
+                driver = new FirefoxDriver();
+                break;
+            case "edge":
+                WebDriverManager.edgedriver().setup();
+                driver = new EdgeDriver();
+                break;
+            case "internetexplorer":
+                WebDriverManager.iedriver().setup();
+                driver = new InternetExplorerDriver();
+                break;
+            default: // Default to Chrome
+                WebDriverManager.chromedriver().setup();
+                driver = new ChromeDriver();
+                break;
+        }
+        maximizeWindow(testProperties.getProperty("url"));
+    }
+
+    private void maximizeWindow(String url) {
```

Comment:
please rename method or remove non related functionality from it

Reply:
Got it, thank you